arangod/IResearch/IResearchAnalyzerFeature.cpp

```diff
@@ -325,143 +325,91 @@ std::string normalizedAnalyzerName(
-    arangodb::UpgradeFeature // feature type
-  >("Upgrade");
+  using namespace arangodb;
+  using namespace arangodb::application_features;
+  using namespace arangodb::methods;
+
+  auto* upgrade = ApplicationServer::lookupFeature<UpgradeFeature>("Upgrade");
 
   if (!upgrade) {
     return; // nothing to register with (OK if no tasks actually need to be applied)
   }
 
-  // register legacy static analyzers with each vocbase found in DatabaseFeature
-  // required for backward compatibility for e.g. TOKENS(...) function
   // NOTE: db-servers do not have a dedicated collection for storing analyzers,
   //       instead they get their cache populated from coordinators
-  {
-    arangodb::methods::Upgrade::Task task;
-    task.name = "IResearhAnalyzer legacy analyzers";
-    task.description = // description
-      "register legacy static analyzers with each vocbase found in DatabaseFeature";
-    task.systemFlag = arangodb::methods::Upgrade::Flags::DATABASE_ALL;
-    task.clusterFlags = // flags
-      arangodb::methods::Upgrade::Flags::CLUSTER_COORDINATOR_GLOBAL // any 1 single coordinator
-      | arangodb::methods::Upgrade::Flags::CLUSTER_NONE // local server
-      ;
-    task.databaseFlags = arangodb::methods::Upgrade::Flags::DATABASE_UPGRADE;
-    task.action = &iresearchAnalyzerLegacyAnalyzers;
-    upgrade->addTask(std::move(task));
-
-    // FIXME TODO find out why CLUSTER_COORDINATOR_GLOBAL will only work with DATABASE_INIT (hardcoded in Upgrade::clusterBootstrap(...))
-    task.name = "IResearhAnalyzer legacy analyzers";
-    task.description =
-      "register legacy static analyzers with each vocbase found in DatabaseFeature";
-    task.systemFlag = arangodb::methods::Upgrade::Flags::DATABASE_ALL;
-    task.clusterFlags = // flags
-      arangodb::methods::Upgrade::Flags::CLUSTER_COORDINATOR_GLOBAL; // any 1 single coordinator
-    task.databaseFlags = arangodb::methods::Upgrade::Flags::DATABASE_INIT;
-    task.action = &iresearchAnalyzerLegacyAnalyzers;
-    upgrade->addTask(std::move(task));
-  }
+
+  upgrade->addTask({
+    "setupAnalyzers",                          // name
+    "setup _analyzers collection",             // description
+    Upgrade::Flags::DATABASE_ALL,              // system flags
+    Upgrade::Flags::CLUSTER_COORDINATOR_GLOBAL // cluster flags
+      | Upgrade::Flags::CLUSTER_NONE,
+    Upgrade::Flags::DATABASE_INIT              // database flags
+      | Upgrade::Flags::DATABASE_UPGRADE
+      | Upgrade::Flags::DATABASE_EXISTING,
+    &setupAnalyzersCollection                  // action
+  });
+
+  upgrade->addTask({
+    "dropLegacyAnalyzersCollection",           // name
+    "drop _iresearch_analyzers collection",    // description
+    Upgrade::Flags::DATABASE_SYSTEM,           // system flags
+    Upgrade::Flags::CLUSTER_COORDINATOR_GLOBAL // cluster flags
+      | Upgrade::Flags::CLUSTER_NONE,
+    Upgrade::Flags::DATABASE_INIT              // database flags
+      | Upgrade::Flags::DATABASE_UPGRADE,
```

Comment:
just wondering why here we don't need to set `Upgrade::Flags::DATABASE_EXISTING`. Do you know?

Reply:
I don't know to be honest, I've just copy-pasted that from upgrade rules that create other system collections, e.g. `_graphs`. I originally thought that `DATABASE_EXISTING` is not necessary either.